helpers/formatHelpers.js

```diff
@@ -0,0 +1,388 @@
+  }
+
+  // Start traversing the path within the data object
+  return checkPathSegment(data, 0);
+}
+
+/**
+ * Converts a numeric value into a suitable one for the unified tracking data
+ *
+ * @param   {*}                numericValue The value that shall be converted
+ * @returns {number|undefined}              The converted value
+ * @private
+ */
+function getUnifiedNumber(numericValue) {
+  // Convert the value
+  const convertedValue = parseFloat(numericValue);
+
+  // Check if the converted value is numeric. If it's not, return "undefined" instead of it
+  return !isNaN(convertedValue) ? convertedValue : undefined;
+}
+
+/**
+ * Returns the productNumber or uid from a product
+ *
+ * @param {Object} product Object with product data
+ * @returns {string} productNumber or uid of the product
+ * @private
+ */
+function getProductIdentifier(product) {
+  return get(product, 'productNumber') || get(product, 'uid');
+}
+
+/**
+ * Removes shop names out of the page title
+ * @param {string} title The page title
+ * @param {string} shopName The shop name
+ * @returns {string} The sanitized page title
+ */
+function sanitizeTitle(title, shopName) {
+  // Take care that the parameters don't contain leading or trailing spaces
+  let trimmedTitle = title.trim();
+  const trimmedShopName = shopName.trim();
+
+  if (!trimmedShopName) {
+    // If no shop name is available, it doesn't make sense to replace it.
+    // So we return the the trimmed title directly.
+    return trimmedTitle;
+  }
+
+  /*
+   Setup the RegExp. It matches leading and trailing occurrences
+   of known patterns for generically added shop names within page title
+   */
+  const shopNameRegExp = new RegExp(`((^${trimmedShopName}:)|(- ${trimmedShopName}$))+`, 'ig');
+
+  if (trimmedTitle === trimmedShopName) {
+    // Clear the page title if it only contains the shop name
+    trimmedTitle = '';
+  }
+
+  // Remove the shop name from the page title
+  return trimmedTitle.replace(shopNameRegExp, '').trim();
+}
+
+/**
+ * Convert sgData products to unified items
+ *
+ * @param {Array} products Items from sgData
+ * @returns {UnifiedPurchaseItem[] | UnifiedAddToCartItem[]} Data for the unified items
+ */
+function formatSgDataProducts(products) {
+  // TODO: Error handling for malformed data
+
+  if (!products || !Array.isArray(products)) {
+    return [];
+  }
+
+  return products.map(product => ({
+    id: getProductIdentifier(product),
+    type: 'product',
+    name: get(product, 'name'),
+    priceNet: getUnifiedNumber(get(product, 'amount.net')),
+    priceGross: getUnifiedNumber(get(product, 'amount.gross')),
+    quantity: getUnifiedNumber(get(product, 'quantity')),
+    currency: get(product, 'amount.currency'),
+  }));
+}
+
+/**
+ * Convert products from the favouriteListItemAdded event to unified items
+ *
+ * @param {Array} products Items from sgData
+ * @returns {UnifiedAddToWishlistItem[]} Data for the unified items
+ */
+function formatFavouriteListItems(products) {
```

Comment:
I would recommend to rename this function and adjust the comment. The function is used within the `addToWishlist` `dataFormatHelper`